### PR TITLE
Add private s3 bucket support

### PR DESCRIPTION
**Pull request by oojo12**
# What does this PR do? Documents private S3 bucket support

Explain purpose of PR and any useful context. Explain to users how to use private S3 buckets

### CODE REVIEW

**Comment by oojo12** [comment]
I'm going to see about making the json text prettier.

File: docs/products/add-data-to-grid-datastores/README.md [hunk 1]
```diff
@@ -17,6 +17,37 @@ Datastores can be attached to Runs or Sessions and preserve the file format and
...
+               "arn:aws:s3:::your-bucket/*",
+               "arn:aws:s3:::your-bucket"
+           ]               
+        }
+    ]
+}
+
+### Known limitations:
+
+Currently, we cannot guarantee the privacy of datastores connected to a public grid cloud account. It is on the roadmap and will be implemented in a future release. However, if you would like to test this feature out you can replace the principal id with 302180240179.
```
Comment (rlizzo) [inline docs/products/add-data-to-grid-datastores/README.md:50]: After second thought, I don't think we should say this publically

> However, if you would like to test this feature out you can replace the principal id with 302180240179

I might say something to the effect of: "there are limited circumstances under which testing of a private aws s3 bucket is possible on the grid public cloud. Please reach out to support@grid.ai if you are interested." that way we can make sure the user ACTUALLY understands the implications here before giving them an account ID which if they used would publically expose their data.

File: docs/products/add-data-to-grid-datastores/README.md [hunk 2]
```diff
@@ -17,6 +17,37 @@ Datastores can be attached to Runs or Sessions and preserve the file format and
 We don't charge for data storage!
 :::
 
+## Datastores with Private S3 Buckets
+
+We only recommend using private S3 buckets for BYOC users. This can be accomplished with the following steps:
```
Comment (rlizzo) [inline docs/products/add-data-to-grid-datastores/README.md]: This isn't a recomendation. It is the only thing we are willing to support / guarantee right now.

**Review by rlizzo — CHANGES_REQUESTED** [review]
A few important techincal changes requested. 

I would also like one of the technical writers to polish this up. The language is very characteristic of my "engineer talk" writing voice. 

@essiequoi can you please delegate as appropriate?

File: docs/products/add-data-to-grid-datastores/README.md [hunk 3]
```diff
@@ -17,6 +17,37 @@ Datastores can be attached to Runs or Sessions and preserve the file format and
...
+1. follow the official aws [documentation](https://aws.amazon.com/premiumsupport/knowledge-center/cross-account-access-s3/).
+2. Use this for the principal field: "Principal": { "AWS": "arn:aws:iam::aws-account-id-associated-with-byoc:root" }
+3. Use the following policy to give access to all the contents within your bucket:
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::302180240179:root"
```
[the remark this answers is not in the document]
Comment (oojo12) [reply docs/products/add-data-to-grid-datastores/README.md]: 1st part has been addressed. I will add some instruction for finding your aws account id.

File: docs/products/add-data-to-grid-datastores/README.md (hunk 1 above)
Comment (oojo12) [reply docs/products/add-data-to-grid-datastores/README.md]: addressed

File: docs/products/add-data-to-grid-datastores/README.md [hunk 4]
```diff
@@ -17,6 +17,37 @@ Datastores can be attached to Runs or Sessions and preserve the file format and
...
+
+As a convenience we provide a bucket policy that grants Grid access to all the contents of your specified bucket.
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::aws-account-id-associated-with-byoc:root"
```
Comment (rlizzo) [inline docs/products/add-data-to-grid-datastores/README.md:32]:
```suggestion
                "AWS": "arn:aws:iam::<aws-account-id-associated-with-byoc>:root"
```

Please enclose in brackets to indicate this field is replaced b the user.

File: docs/products/add-data-to-grid-datastores/README.md [hunk 5]
```diff
@@ -17,6 +17,37 @@ Datastores can be attached to Runs or Sessions and preserve the file format and
...
+               "arn:aws:s3:::your-bucket"
+           ]               
+        }
+    ]
+}
+```
+
+### Known limitations:
+
+Currently, we cannot guarantee the privacy of datastores connected to a public grid cloud account. It is on the roadmap and will be implemented in a future release. There are limited circumstances under which testing of a private aws s3 bucket is possible on the grid public cloud. Please reach out to support@grid.ai if you are interested.
```
Comment (rlizzo) [inline docs/products/add-data-to-grid-datastores/README.md:50]: > Currently, we cannot guarantee the privacy of datastores connected to a public grid cloud account. It is on the roadmap and will be implemented in a future release. There are limited circumstances under which testing of a private aws s3 bucket is possible on the grid public cloud.

This sounds clunky. Let's get some input from the docs writers

**Review by rlizzo — CHANGES_REQUESTED** [review]

File: docs/products/add-data-to-grid-datastores/README.md (hunk 5 above)
Comment (oojo12) [reply docs/products/add-data-to-grid-datastores/README.md:50]: sounds good. @essiequoi has been requested to review everything as well.

File: docs/products/add-data-to-grid-datastores/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (oojo12) [reply docs/products/add-data-to-grid-datastores/README.md]: 2nd part addressed with official documentation.

**Review by nmiculinic — APPROVED** [review]

File: docs/products/add-data-to-grid-datastores/README.md [hunk 6]
```diff
@@ -17,7 +17,33 @@ Datastores can be attached to Runs or Sessions and preserve the file format and
...
+
+As a convenience, below we provided a bucket policy that grants Grid access to all the contents of your specified bucket. You can follow this official aws [documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/console_account-alias.html) to get your account id.
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::<aws-account-id-associated-with-byoc>:root"
```
Comment (nmiculinic) [inline docs/products/add-data-to-grid-datastores/README.md:32]: Hmm...technically here we can be more precise and directly specify the role for this cluster-id which will be doing datastore copying?

Comment (oojo12) [reply docs/products/add-data-to-grid-datastores/README.md:32]: and if you didn't specify one use root

File: docs/products/add-data-to-grid-datastores/README.md (hunk 5 above)
Comment (essiequoi) [reply docs/products/add-data-to-grid-datastores/README.md]: I have a proposition for a better way to phrase this, but the current question is whether we should even mention this at all or if we should restrict the use of this feature to BYOC users. We're discussing it on the Linear ticket, will report back

Comment (essiequoi) [reply docs/products/add-data-to-grid-datastores/README.md]: i see this was removed, so ignore the above comment

**Review by essiequoi — CHANGES_REQUESTED** [review]
one change requested. the rest is approved from me

File: docs/products/add-data-to-grid-datastores/README.md [hunk 7]
```diff
@@ -21,7 +21,7 @@ We don't charge for data storage!
 
 At this time we are only supporting usage of private S3 buckets as datastores for BYOC users. You can grant Grid access to your desired buckets by following the official aws [documentation](https://aws.amazon.com/premiumsupport/knowledge-center/cross-account-access-s3/).
```
[the remark this answers is not in the document]
Comment (oojo12) [reply docs/products/add-data-to-grid-datastores/README.md]: addressed.

**Pull request merged**